### PR TITLE
feat: Support for IndexedDB

### LINKED ISSUE #809: Support for IndexedDB
Wrap as Go functions that implements storage interface (pkg/storage/indexeddb)

part of #661

### PR DESCRIPTION
**Pull request by fqutishat**
closes #809

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 GO_CMD ?= go
 ARIES_AGENT_REST_PATH=cmd/aries-agent-rest
+ARIES_AGENT_WASM_PATH=cmd/aries-agent-wasm
```
Comment (troyronda) [inline Makefile]: I wonder if this is `ARIES_AGENT_WASM` or simply `ARIES_WASM`

File: cmd/aries-agent-wasm/main.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	}
+
+	fmt.Println("Creating invitation")
+
+	i, err := c.CreateInvitation("foo")
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Printf("ID: %s; RecipientKeys: %s\n", i.ID, i.RecipientKeys)
```
Comment (troyronda) [inline cmd/aries-agent-wasm/main.go]: I assume you are removing these PoC lines eventually?

File: pkg/framework/aries/default_srv.go [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+*/
+
+package aries
+
+import (
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+	"github.com/hyperledger/aries-framework-go/pkg/storage/leveldb"
+)
+
+func storeProvider() (storage.Provider, error) {
```
Comment (troyronda) [inline pkg/framework/aries/default_srv.go]: TODO:

- rename transient store to protocol state store (or similar).
- use indexedDB (as default) with a JS target.

(cc: @rolsonquadras )

File: cmd/aries-agent-wasm/main.go (hunk 2 above)
Comment (fqutishat) [reply cmd/aries-agent-wasm/main.go]: yes

File: pkg/framework/aries/default_srv.go (hunk 3 above)
Comment (fqutishat) [reply pkg/framework/aries/default_srv.go]: I add TODO for rename transient store to protocol state storehttps://github.com/hyperledger/aries-framework-go/pull/830/files#diff-7b2448fdc2181299533bc944239b9744R28

Comment (fqutishat) [reply pkg/framework/aries/default_srv.go]: use indexedDB (as default) with a JS target.  (DONE)

**Comment by troyronda** [comment]
Note: tests will be coming in an upcoming PR soon.

**Pull request merged**